compiler/rustc_resolve/src/late.rs

```diff
@@ -3341,8 +3341,20 @@ impl<'a: 'ast, 'b, 'ast> LateResolutionVisitor<'a, 'b, 'ast> {
 
                 let def_id = this.parent_scope.module.nearest_parent_mod();
                 let instead = res.is_some();
-                let suggestion =
-                    if res.is_none() { this.report_missing_type_error(path) } else { None };
+                let suggestion = if let Some((start, end)) = this.diagnostic_metadata.in_range
+                    && path[0].ident.span.lo() == end.span.lo()
```

Comment:
I didn't look too much into this, but does this check that this is a method?

We should also probably double check this is a path with one segment, right?

Reply:
It doesn't, but during resolve we don't have enough data to do that (although now that I'm tracking `start` I *might* be able to cobble something together).

Reply:
I'm comfortable with the weasel wording on this one suggestion: it is a guess, and we'd have to perform more extensive surgery on resolve to track more state, and it is already quite a beast.

Reply:
My only concern is that this triggers when we have any resolution error on the RHS of a range, e.g.

```rust
fn foo() {
  a..b;
}
```

Results in:
```
error[E0425]: cannot find value `a` in this scope
 --> /home/ubuntu/test.rs:2:5
  |
2 |     a..b;
  |     ^ not found in this scope

error[E0425]: cannot find value `b` in this scope
 --> /home/ubuntu/test.rs:2:8
  |
2 |     a..b;
  |        ^ not found in this scope
  |
help: you might have meant to write a method call instead of a range
  |
2 -     a..b;
2 +     a.b;
```

I'd much prefer if the wording was made to not mention methods in specific, but focus on the `.` itself -- something like "you might've meant to write the `.` operator instead of a range"?

Reply:
What do you think of "you might have meant to write `.` instead of `..`"

Reply:
Yeah, that works fine. 

Just would prefer it to not mention "methods" unless we can verify it's actually a method, but since this code is in resolve, I totally understand how that's hard, haha.